changelog/583.breaking.rst

```diff
@@ -0,0 +1,3 @@
+Update :class:`ModalInteraction` typings.
```

Comment:
Do we want to add a feature fragment for `ModalInteractionData.components` as well?